### PR TITLE
Decorator syntax support

### PR DESCRIPTION
**Pull request by xen**
Just playing with Sanic and found your extension. I've added decorator method that makes more convenient way to work with rendering templates. I didn't make any tests yet. It is a quick hack, but I plan to work with extension in the future so, probably if it misses something I will add it.

### CODE REVIEW

**Comment by lixxu** [comment]
did you try it to see it's work or not?

**Comment by xen** [comment]
I checked on my project for simple cases.

File: sanic_jinja2/__init__.py [hunk 1]
```diff
@@ -105,3 +127,44 @@ def _get_flashed_messages(self, request, with_categories=False,
...
+
+                if isinstance(context, HTTPResponse):
+                    return context
+
+                if isinstance(args[0], HTTPMethodView):
+                    request = args[1]
+                else:
+                    request = args[0]
+
+                return request.app[APP_KEY].render(template_name, request, context,
```
Comment (lixxu) [inline sanic_jinja2/__init__.py]: `request.app` is dict-like? I can't see it in sanic source code.

File: sanic_jinja2/__init__.py [hunk 2]
```diff
@@ -2,18 +2,26 @@
...
+import functools
+from sanic.response import html, HTTPResponse
+from sanic.views import HTTPMethodView
 from jinja2 import Environment, PackageLoader
 from jinja2.ext import _make_new_gettext, _make_new_ngettext
 
 __version__ = '0.5.5'
 
+CONTEXT_PROCESSORS = 'context_processor'
+APP_KEY = 'jinja2_env'
```
Comment (lixxu) [inline sanic_jinja2/__init__.py]: what's the purpose to define `APP_KEY` here? If its purpose is allows user to customize the name, it should not be here.

File: sanic_jinja2/__init__.py [hunk 3]
```diff
@@ -64,32 +77,41 @@ def update_request_context(self, request, context):
...
+
     def render_string(self, template, request, **context):
         self.update_request_context(request, context)
         return self.env.get_template(template).render(**context)
 
+
     def render(self, template, request, **context):
         return html(self.render_string(template, request, **context))
 
+
```
Comment (lixxu) [inline sanic_jinja2/__init__.py]: why put two newlines in class methods?

Comment (xen) [reply sanic_jinja2/__init__.py]: It was settings for my editor. I'll fix it.

**Comment by xen** [comment]
I've checked pull request again, by mistake, I tested against wrong code. Sort things out.

File: sanic_jinja2/__init__.py [hunk 4]
```diff
@@ -28,8 +75,14 @@ def init_app(self, app, loader=None, pkg_name=None, pkg_path=None):
...
             app.extensions = {}
 
+        if self.context_processors:
+            if not hasattr(app, CONTEXT_PROCESSORS):
+                setattr(app, CONTEXT_PROCESSORS, self.context_processors)
+                app.request_middleware.append(self.context_processors)
+
         app.extensions['jinja2'] = self
-        app.jinja_env = self.env
+        app.jinja2 = self.env
```
Comment (lixxu) [inline sanic_jinja2/__init__.py]: this will break the code that uses the old version. If you want to change the name, please pass it via the argument.

**Comment by xen** [comment]
Please, review again.

**Pull request merged**